nova/scheduler/weights/ram.py

```diff
@@ -36,3 +36,24 @@ def weight_multiplier(self):
     def _weigh_object(self, host_state, weight_properties):
         """Higher weights win.  We want spreading to be the default."""
         return host_state.free_ram_mb
+
+
+class RamResizeBigVmToSameHostWeigher(weights.BaseHostWeigher):
+    minval = 0
+
+    def weight_multiplier(self):
+        """Override the weight multiplier."""
+        return CONF.filter_scheduler.ram_resize_to_same_host_weight_multiplier
+
+    def _weigh_object(self, host_state, weight_properties):
+        """Return 1 for big VMs when the host is the instances current host.
+        Return 0 otherwise.
+        """
+        instance_uuid = weight_properties.instance_uuid
+        if instance_uuid in host_state.instances:
+            instance = host_state.instances[instance_uuid]
+            ram_before = instance.old_flavor.memory_mb
+            ram_after = instance.new_flavor.memory_mb
+            if max(ram_before, ram_after) >= CONF.bigvm_mb:
+                return 1.0
```

Comment:
Please use the `is_big_vm()` function from `nova.utils` if possible. Or rather don't use it, because we want this functionality for all VMs.